Sources/SwiftCentrifuge/Client.swift

```diff
@@ -532,11 +556,48 @@ fileprivate extension CentrifugeClient {
         assertIsOnQueue(syncQueue)
 
         guard self.state == .connecting else { return }
-
-        if refreshRequired || (token == "" && config.tokenGetter != nil) {
-            getConnectionToken(completion: { [weak self] result in
-                guard let strongSelf = self, strongSelf.state == .connecting else { return }
-                switch result {
+        if refreshRequired ||
+            (data == nil && config.dataGetter != nil) ||
+            (!token.isEmpty && config.tokenGetter != nil) {
+            let updateData = config.dataGetter != nil
+            let updateToken = config.tokenGetter != nil
+            if updateData {
```

Comment:
I do not see where token is updated in this case. Generally, I think to refactor code a bit for this feature to make the code easier to follow. It already was a bit hard to follow.